### ISSUE TITLE
Workflow with external genomes

### ISSUE THREAD

**Issue by chatcrawler** [ODoSE/odose.nl#3]
Hi Tim.

It seems that I cannot finish any analyses based solely on external genomes with ODoSE (correct me where I am wrong), and I think this is mainly due to the fact that one cannot select any published GenBank genomes in ODoSE anymore.

More specifically: when I upload all my external genomes in nucleotide *fasta-file-format (as described in the main publication; Vos et al. 2013 PLoS ONE), I don't find a way to select these and translate their genes into amino acid sequences prior to the orthoMCL step (which is based on protein-BLAST searches, blastp).

However, when I opt to upload amino acid *fasta-files in the first place, I also get stuck, this time right AFTER the orthoMCL step. The latter runs through efficiently, but in order to execute the "Extract & categorize orthologs from OrthoMCL groups file" step I would have to choose *GenBank Project IDs* which were not created in the first place (in the absence of selectable gbk-files). And thus the whole analysis lead to nonsensical output, e.g., no SICOs are reported and there are as many genomes as genes declared in the output.

For the moment I don't see a way out of this, please let me know, what you would suggest me to do.

Cheers

Séb

**Comment by chatcrawler** [comment]
Small addition to what I have just described:

my *genomes* are not closed but are stored in the format of multi-sequence fasta files:

>Gene 1
GATC
>Gene 2
ATCG

...

**Comment by timtebeek** [comment]
Hi @chatcrawler ,
Allow me to copy what I also sent in an email earlier, in hopes that it helps, or at least helps pinpoint the exact problem:
> The functionality [to run external genomes] is still there! It's just switched around a bit since the Galaxy devs broke the old way of side-loading external genomes. The manual still needs to be posted to the website, but here's the gist:
- Start a new history
- Drag (DNA) fasta files into your history, or upload using the Icon top left, or with `Get Data > Upload File`. Be sure to set the type/extension to fasta when you do.
- Select the tool `Ortholog Direction of Selection Engine > Upload genomes`
- For each of your uploaded fasta files, provide a short (<10) unique label.
- Execute the tool once you've added all external genomes, this should result in a single new history item.
- At workflow `Step 2: Extract & translate genes`, select your new history item of externally uploaded genomes
- Continue as before.

**Comment by chatcrawler** [comment]
Hi Tim.

Thanks a lot for your efforts yet again.

I have performed your described steps to the comma:
- Start a new history
- Drag (DNA) fasta files into your history, or upload using the Icon top left, or with `Get Data > Upload File`. Be sure to set the type/extension to fasta when you do.
- Select the tool `Ortholog Direction of Selection Engine > Upload genomes`
- For each of your uploaded fasta files, provide a short (<10) unique label.
- Execute the tool once you've added all external genomes, this should result in a single new history item.
- At workflow `Step 2: Extract & translate genes`, select your new history item of externally uploaded genomes

But the translation step does not work! I have attached two snapshots from my Browser window:

>Snap-Execute.png

![snap-execute](https://cloud.githubusercontent.com/assets/2347378/10850175/82c0adfc-7f24-11e5-8089-7ec69f176412.png)
Shows you the view on the Drop-Down menu and highlights the issue, I described in my earlier eMail, namely, that I am forced to select either a GenBank genome (or one of my 22 single genomes in fasts-file format) in the first of two DropDown-fields. Thus, I had to select something (I cannot select nothing) and chose A93.ffn. Now only in the second optional DropDown field for External Genomes can I choose the history item *External Genomes*.

>Snap-Result.png

![snap-result](https://cloud.githubusercontent.com/assets/2347378/10850198/98e4be0c-7f24-11e5-872e-dfa4c3dfd42f.png)
After I executed the command, I get an error message which related back to the fact that I chose A93.ffn:

Could not find genome with BioProject ID >MXA93_01952 in complete genomes table
Could not find genome with BioProject ID >MXA93_01953 in complete genomes table
Could not find genome with BioProject ID >MXA93_01954 in complete genomes table
Could not find genome with BioProject ID >MXA93_01955 in complete genomes table
Could not find genome with BioProject ID >MXA93_01956 in complete genomes table
Could not find genome with BioProject ID >MXA93_01957 in complete genomes table
Could not find genome with BioProject ID >MXA93_01958 in complete genomes table
Could not find genome with BioProject ID >MXA93_01959 in complete genomes table
Could not find genome with BioProject ID >MXA93_01960 in complete genomes table
Could not find genome with BioProject ID >MXA93_01961 in complete genomes table
Could not find genome with BioProject ID >MXA93_01962 in complete genomes table
Could not find genome with BioProject ID >MXA93_01963 in complete genomes table
Could not find genome with BioProject ID >MXA93_01964 in complete genomes table
Could not find genome with BioProject ID >MXA93_01965 in complete genomes table
Could not find genome with BioProject ID >MXA93_01966 in complete genomes table
Could not find genome with BioProject ID >MXA93_01967 in complete genomes table
Could not find genome with BioProject ID >MXA93_01968 in complete genomes table
Could not find genome with BioProject ID >MXA93_01969 in complete genomes table
Could not find genome with BioProject ID >MXA93_01970 in complete genomes table
Could not find genome with BioProject ID >MXA93_01971 in complete genomes table
Could not find genome with BioProject ID >MXA93_01972 in complete genomes table
Could not find genome with BioProject ID >MXA93_01973 in complete genomes table
Could not find genome with BioProject ID >MXA93_01974 in complete genomes table
Could not find genome with BioProject ID >MXA93_01975 in complete genomes table
Could not find genome with BioProject ID >MXA93_01976 in complete genomes table
Could not find genome with BioProject ID >MXA93_01977 in complete genomes table
Could not find genome with BioProject ID >MXA93_01978 in complete genomes table
Could not find genome with BioProject ID >MXA93_01979 in complete genomes table
Could not find genome with BioProject ID >MXA93_01980 in complete genomes table
Could not find genome with BioProject ID >MXA93_01981 in complete genomes table
Could not find genome with BioProject ID >MXA93_01982 in complete genomes table
Could not find genome with BioProject ID >hsdR in complete genomes table
Could not find genome with BioProject ID >MXA93_01984 in complete genomes table
Could not find genome with BioProject ID >MXA93_01985 in complete genomes table
Could not find genome with BioProject ID >MXA93_01986 in complete genomes table
Could not find genome with BioProject ID >MXA93_01987 in complete genomes table
Could not find genome with BioProject ID >MXA93_01988 in complete genomes table
Could not find genome with BioProject ID >MXA93_01989 in complete genomes table
Could not find genome with BioProject ID >MXA93_01990 in complete genomes table
Could not find genome with BioProject ID >MXA93_01991 in complete genomes table
Could not find genome with BioProject ID >MXA93_01992 in complete genomes table
Could not find genome with BioProject ID >MXA93_01993 in complete genomes table
Could not find genome with BioProject ID >MXA93_01994 in complete genomes table
Could not find genome with BioProject ID >MXA93_01995 in complete genomes table
Could not find genome with BioProject ID >MXA93_01996 in complete genomes table
Could not find genome with BioProject ID >MXA93_01997 in complete genomes table
Could not find genome with BioProject ID >MXA93_01998 in complete genomes table
Could not find genome with BioProject ID >MXA93_01999 in complete genomes table
Could not find genome with BioProject ID >MXA93_02000 in complete genomes table
Could not find genome with BioProject ID >MXA93_02001 in complete genomes table
Could not find genome with BioProject ID >MXA93_02002 in complete genomes table
Could not find genome with BioProject ID >MXA93_02003 in complete genomes table
Could not find genome with BioProject ID >MXA93_02004 in complete genomes table
Could not find genome with BioProject ID >MXA93_02005 in complete genomes table
Could not find genome with BioProject ID >MXA93_02006 in complete genomes table
Could not find genome with BioProject ID >MXA93_02007 in complete genomes table
Could not find genome with BioProject ID >MXA93_02008 in complete genomes table
Could not find genome with BioProject ID >MXA93_02009 in complete genomes table
Could not find genome with BioProject ID >MXA93_02010 in complete genomes table
Could not find genome with BioProject ID >MXA93_02011 in complete genomes table
Could not find genome with BioProject ID >MXA93_02012 in complete genomes table
Could not find genome with BioProject ID >MXA93_02013 in complete genomes table
Could not find genome with BioProject ID >MXA93_02014 in complete genomes table
Could not find genome with BioProject ID >MXA93_02015 in complete genomes table
Could not find genome with BioProject ID >MXA93_02016 in complete genomes table
Could not find genome with BioProject ID >MXA93_02017 in complete genomes table
Could not find genome with BioProject ID >MXA93_02018 in complete genomes table
Could not find genome with BioProject ID >MXA93_02019 in complete genomes table
Could not find genome with BioProject ID >MXA93_02020 in complete genomes table
Could not find genome with BioProject ID >MXA93_02021 in complete genomes table
Could not find genome with BioProject ID >MXA93_02022 in complete genomes table
Could not find genome with BioProject ID >MXA93_02023 in complete genomes table
Could not find genome with BioProject ID >MXA93_02024 in complete genomes table
Could not find genome with BioProject ID >MXA93_02025 in complete genomes table
Could not find genome with BioProject ID >MXA93_02026 in complete genomes table
Could not find genome with BioProject ID >MXA93_02027 in complete genomes table
Could not find genome with BioProject ID >MXA93_02028 in complete genomes table
Could not find genome with BioProject ID >MXA93_02029 in complete genomes table
Could not find genome with BioProject ID >MXA93_02030 in complete genomes table
Could not find genome with BioProject ID >MXA93_02031 in complete genomes table
Could not find genome with BioProject ID >MXA93_02032 in complete genomes table
Could not find genome with BioProject ID >MXA93_02033 in complete genomes table
Could not find genome with BioProject ID >MXA93_02034 in complete genomes table
Could not find genome with BioProject ID >MXA93_02035 in complete genomes table
Could not find genome with BioProject ID >MXA93_02036 in complete genomes table
Could not find genome with BioProject ID >MXA93_02037 in complete genomes table
Could not find genome with BioProject ID >MXA93_02038 in complete genomes table
Could not find genome with BioProject ID >MXA93_02039 in complete genomes table
Could not find genome with BioProject ID >MXA93_02040 in complete genomes table
Could not find genome with BioProject ID >MXA93_02041 in complete genomes table
Could not find genome with BioProject ID >MXA93_02042 in complete genomes table
Could not find genome with BioProject ID >MXA93_02043 in complete genomes table
Could not find genome with BioProject ID >MXA93_02044 in complete genomes table
Could not find genome with BioProject ID >MXA93_02045 in complete genomes table
Could not find genome with BioProject ID >MXA93_02046 in complete genomes table
Could not find genome with BioProject ID >MXA93_02047 in complete genomes table
Could not find genome with BioProject ID >MXA93_02048 in complete genomes table
Could not find genome with BioProject ID >MXA93_02049 in complete genomes table
Could not find genome with BioProject ID >MXA93_02050 in complete genomes table
Could not find genome with BioProject ID >MXA93_02052 in complete genomes table
Could not find genome with BioProject ID >MXA93_02053 in complete genomes table
Could not find genome with BioProject ID >MXA93_02054 in complete genomes table
Could not find genome with BioProject ID >MXA93_02055 in complete genomes table
Could not find genome with BioProject ID >MXA93_02056 in complete genomes table
Could not find genome with BioProject ID >suhB in complete genomes table
Could not find genome with BioProject ID >MXA93_02058 in complete genomes table
Could not find genome with BioProject ID >MXA93_02059 in complete genomes table
Could not find genome with BioProject ID >MXA93_02060 in complete genomes table
Could not find genome with BioProject ID >MXA93_02061 in complete genomes table
Could not find genome with BioProject ID >MXA93_02062 in complete genomes table
An assumption failed
Traceback (most recent call last):
  File "/work/odosenl/galaxy-dist/tools/odosenl/wrapper.py", line 32, in <module>
    MODULE.main(sys.argv[3:])
  File "/work/odosenl/odose.nl/translate.py", line 358, in main
    dna_files, protein_files = translate_genomes(genomes)
  File "/work/odosenl/odose.nl/translate.py", line 81, in translate_genomes
    assert len(genomes), 'Some genomes should be selected'
AssertionError: Some genomes should be selected


Now I can circumvent that issue by simply uploading the same genomes but now in amino acid fasta file format: I would't have to translate them into amino acids using your Tool and thus can go ahead with the OrthoMCL step. However, then I get stuck when I want to *Extract & Categorize orthologs* because ODoSE does create and ID file of all genomes, and without it that latter step fails. Since I do not know how that file looks like, I cannot add it externally.


Cheers and many thanks again.


Séb

**Comment by timtebeek** [comment]
Hi Seb,

I must have gotten a minor detail wrong! Nadia (cc) is our most experienced
odose user at the moment, and demonstrated yesterday that she's able to do
runs with only external genomes.

Nadia, do you think you can fill me and Séb in on how you managed to run
external genomes only? We'll update the manual to match once you do!

Best,
Tim

**Comment by timtebeek** [comment]
Hi guys...
here I am!
here is my workflow to work with external genomes... I am not sure if it's
the right way to use it, but it should work!

first of all., I work with external genomes downloaded from ncbi in the ffn
format.
then:

1. load selectable genomes>execute
2. upload your own externals using the up arrow on the grey bar next to the
word "TOOLS" (.ffn files in my case... remember to select the format)>start
3.upload genomes (I usually give them the number of the upload to avoid
long names) selecting each file and giving  an univoque name>execute
4. then I usually start a workflow that Michiel Vos gave me! (
http://www.odose.nl/u/nadiaandreani/w/imported-imported-odose-without-inter-taxon-recombination-filtering-1
  <http://www.odose.nl/workflow/sharing?id=843d7cb7925ea10e#>) just import
it if you need it! in the workflow I select "selectable genomes" at point1
and external genomes (that I just created when uploading my files at step
3) just below.

I tried yesterday with 5 acinetobacter baumanii external genomes and it was
working! at the moment 4 runs with 25 external genomes are on (OrthoMCL
step is on at the moment for all of them...). I will let you know if it's
working properly!

I really hope to have been useful, as I am not a bioinformatics expert but
just an assiduous user!

Let's keep in touch if you need any other info or if I was not clear!


Nadia

2015-10-31 14:35 GMT+01:00 Tim te Beek <timtebeek@gmail.com>:

> Hi Seb,
>
> I must have gotten a minor detail wrong! Nadia (cc) is our most
> experienced odose user at the moment, and demonstrated yesterday that she's
> able to do runs with only external genomes.
>
> Nadia, do you think you can fill me and Séb in on how you managed to run
> external genomes only? We'll update the manual to match once you do!
>
> Best,
> Tim
> On 30 Oct 2015 16:38, "chatcrawler" <notifications@github.com> wrote:
>
>> Hi Tim.
>>
>> Thanks a lot for your efforts yet again.
>>
>> I have performed your described steps to the comma:
>>
>>    - Start a new history
>>    - Drag (DNA) fasta files into your history, or upload using the Icon
>>    top left, or with Get Data > Upload File. Be sure to set the
>>    type/extension to fasta when you do.
>>    - Select the tool Ortholog Direction of Selection Engine > Upload
>>    genomes
>>    - For each of your uploaded fasta files, provide a short (<10) unique
>>    label.
>>    - Execute the tool once you've added all external genomes, this
>>    should result in a single new history item.
>>    - At workflow Step 2: Extract & translate genes, select your new
>>    history item of externally uploaded genomes
>>
>> But the translation step does not work! I have attached two snapshots
>> from my Browser window:
>>
>> Snap-Execute.png
>>
>> [image: snap-execute]
>> <https://cloud.githubusercontent.com/assets/2347378/10850175/82c0adfc-7f24-11e5-8089-7ec69f176412.png>
>> Shows you the view on the Drop-Down menu and highlights the issue, I
>> described in my earlier eMail, namely, that I am forced to select either a
>> GenBank genome (or one of my 22 single genomes in fasts-file format) in the
>> first of two DropDown-fields. Thus, I had to select something (I cannot
>> select nothing) and chose A93.ffn. Now only in the second optional DropDown
>> field for External Genomes can I choose the history item *External
>> Genomes*.
>>
>> Snap-Result.png
>>
>> [image: snap-result]
>> <https://cloud.githubusercontent.com/assets/2347378/10850198/98e4be0c-7f24-11e5-872e-dfa4c3dfd42f.png>
>> After I executed the command, I get an error message which related back
>> to the fact that I chose A93.ffn:
>>
>> Could not find genome with BioProject ID >MXA93_01952 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01953 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01954 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01955 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01956 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01957 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01958 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01959 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01960 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01961 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01962 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01963 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01964 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01965 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01966 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01967 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01968 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01969 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01970 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01971 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01972 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01973 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01974 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01975 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01976 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01977 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01978 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01979 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01980 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01981 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01982 in complete genomes
>> table
>> Could not find genome with BioProject ID >hsdR in complete genomes table
>> Could not find genome with BioProject ID >MXA93_01984 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01985 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01986 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01987 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01988 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01989 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01990 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01991 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01992 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01993 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01994 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01995 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01996 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01997 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01998 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_01999 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02000 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02001 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02002 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02003 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02004 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02005 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02006 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02007 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02008 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02009 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02010 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02011 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02012 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02013 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02014 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02015 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02016 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02017 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02018 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02019 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02020 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02021 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02022 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02023 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02024 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02025 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02026 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02027 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02028 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02029 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02030 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02031 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02032 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02033 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02034 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02035 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02036 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02037 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02038 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02039 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02040 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02041 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02042 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02043 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02044 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02045 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02046 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02047 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02048 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02049 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02050 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02052 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02053 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02054 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02055 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02056 in complete genomes
>> table
>> Could not find genome with BioProject ID >suhB in complete genomes table
>> Could not find genome with BioProject ID >MXA93_02058 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02059 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02060 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02061 in complete genomes
>> table
>> Could not find genome with BioProject ID >MXA93_02062 in complete genomes
>> table
>> An assumption failed
>> Traceback (most recent call last):
>> File "/work/odosenl/galaxy-dist/tools/odosenl/wrapper.py", line 32, in
>> MODULE.main(sys.argv[3:])
>> File "/work/odosenl/odose.nl/translate.py", line 358, in main
>> dna_files, protein_files = translate_genomes(genomes)
>> File "/work/odosenl/odose.nl/translate.py", line 81, in translate_genomes
>> assert len(genomes), 'Some genomes should be selected'
>> AssertionError: Some genomes should be selected
>>
>> Now I can circumvent that issue by simply uploading the same genomes but
>> now in amino acid fasta file format: I would't have to translate them into
>> amino acids using your Tool and thus can go ahead with the OrthoMCL step.
>> However, then I get stuck when I want to *Extract & Categorize orthologs*
>> because ODoSE does create and ID file of all genomes, and without it that
>> latter step fails. Since I do not know how that file looks like, I cannot
>> add it externally.
>>
>> Cheers and many thanks again.
>>
>> Séb
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/ODoSE/odose.nl/issues/3#issuecomment-152559042>.
>>
>


-- 
Nadia Andrea Andreani
PhD Student

Doctoral School in Veterinary Science
Department of Comparative Biomedicine and Food Science (BCA)
University of Padova
Viale dell'Università 16-35020, Legnaro (PD), Italy

**Comment by chatcrawler** [comment]
Hi Nadia.

Thanks so much for your fast reply. I was able to import the workflow. 
For the moment, I am unable to test it since the server put my jobs on hold. I will let you know what I got!

Cheers

Séb

**Comment by JudithZimmermann** [comment]
Hi guys,

I ran into the exact same problem than Séb! I am also using solely external genomes and could never figure out how to get a Project ID list when I wanted to use only my genomes. I was very happy to see Nadias workflow, who simply selects "Selectable genomes" in the "Select genomes" step, but does not mark any of them (at least thats what I did). Doing this I successfully managed to complete the step of identification of the single set of orthologs! However, I get again an error message when I want to continue with my files in the "Extract & categorize orthologs" step. 
The error message "Could not find genome with BioProject ID XXX in complete genomes table..". 
Nadia, did you ever run into this problem too? I would be very happy for any suggestions how to move on.

Cheers Judith
